flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/windows.scala

```diff
@@ -91,6 +91,10 @@ class OverWindowWithOrderBy(
     */
   def as(alias: Expression): OverWindow = {
 
+    if (null == preceding) {
```

Comment:
For the strict system type, I recommend adding the `OverWindowWithPreceding` class to guide the user to do the correct operation.